docs/creating-a-new-v3-ea.md

```diff
@@ -0,0 +1,655 @@
+
+```typescript
+export const endpoint = new AdapterEndpoint<EndpointTypes>({
+  ...
+  customInputValidation(req: RequestParams, config: AdapterConfig<typeof customSettings>): AdapterError | undefined => {
+    if (req.value && (req.value < 0 || req.value > 100)) {
+      return new AdapterInputError({
+        statusCode: 400,
+        message: 'Value is less than 0 or greater than 100',
+      })
+    }
+    return
+  },
+  ...
+})
+```
+
+### Price Endpoint
+
+As mentioned in the [Adapter Section](#adapter), `PriceAdapter` requires that one of its endpoints is a `PriceEndpoint`. To help standardize the format of a price feed endpoint, the framework provides helpful built-in types. This would alter some of the examples shown in the previous sub-sections like below.
+
+```typescript
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+type EndpointTypes = {
+  // The PriceEndpointInputParameters type is the most common request params for a price endpoint. {base: string, quote: string}
+  Request: {
+    Params: PriceEndpointInputParameters
+  }
+  ...
+  // The SingleNumberResultResponse type is the most common response format for price endpoints. {result: number, data: { result: number }}
+  Response: SingleNumberResultResponse,
+  ...
+}
+
+export const endpoint = new PriceEndpoint<EndpointTypes>({
+  ...
+})
+```
+
+## Transports
+
+A transport is the method in which the data is retrieved from a data provider for that particular endpoint. As seen in the [Endpoints](#endpoints) section, every endpoint requires one and only one transport. Define this transport in the same file as its associated endpoint. To learn more about specific transports, check the [Transport Types](#transport-types) section.
+
+### Routing Transport
+
+As mentioned above, each endpoint only accepts a single transport. However, there are scenarios where a data provider will have multiple ways to access the same data. For example, a data provider could offer crypto data through REST API and Websockets which would require two different transports. The `RoutingTransport` allows an endpoint to have access to both.
+
+The `RoutingTransport` is used to direct requests to different transports based on request parameters and/or adapter configs. In the example above, the requests could be directed to either an HTTP or Websocket transport. An example of what the code would look like for this scenario is below. The code assumes that requests are directed based on an env var, `WS_ENABLED`.
+
+```typescript
+import { httpTransport } from './crypto-http'
+import { wsTransport } from './crypto-ws'
+
+export const routingTransport = new RoutingTransport<EndpointTypes>(
+  {
+    WS: wsTransport,
+    REST: httpTransport,
+  },
+  (req, adapterConfig) => (adapterConfig.WS_ENABLED ? 'WS' : 'REST'),
```

Comment:
While not ready yet, just want to flag that this will be different before end of February.

Reply:
I was thinking about that when adding this. For now, I'll add a `Subject to change` disclaimer